app/infrastructure/external/email/oauth_drivers.py

```
@@ -0,0 +1,316 @@
+                data={
+                    "client_id": self.client_id,
+                    "client_secret": self.client_secret,
+                    "refresh_token": refresh_token,
+                    "grant_type": "refresh_token",
+                },
+            )
+            if response.status_code != 200:
+                logger.error("%s token refresh failed: %s", self.provider_name, response.text)
+                raise ValueError(
+                    f"Token refresh failed: {response.status_code} {response.text}"
+                )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)